### ISSUE TITLE
[Feature Request] 

### ISSUE THREAD

**Issue by guglovich** [telegramdesktop/tdesktop#24594]
### Is your feature request related to a problem?

Yes

### Describe the solution you'd like

I would like to see a folder selected for cache, similar to how a folder is selected for downloading files. This option is even slightly more important than the cache folder. 
The point is that users have different PC configurations:
- HDD /root 
and therefore the slowest speed
- SSD /root 
average performance and waste of flash resource
- Nvme SSD /root
maximum performance, long rewrite life

The first 2 categories should have a choice since there is something to lose.
There is also a choice for them (and me too) in the form of faster RAM, which is also without resource. 

### Describe alternatives you've considered

As a last resort, a configurable config file without crypt.

**Comment by ilya-fedin** [comment]
Well, create a symlink for ~/.local/share/TelegramDesktop and move whenever you want?

**Comment by ilya-fedin** [comment]
> * SSD /root
>   average performance and waste of flash resource
> 
> * Nvme SSD /root
>   maximum performance, long rewrite life

How the data protocol (SATA/NVMe) influences rewrite life? :thinking:

**Comment by guglovich** [comment]
> How the data protocol (SATA/NVMe) influences rewrite life? thinking

The difference is in the years of SSD release and the peaks in popularity. Older SSDs of course have a shorter lifespan. 


> Well, create a symlink for ~/.local/share/TelegramDesktop and move whenever you want?

It is possible, of course. But if it was by UX and for everyone, I wouldn't be writing here.

**Comment by ilya-fedin** [comment]
> The difference is in the years of SSD release and the peaks in popularity. Older SSDs of course have a shorter lifespan.

Well, there are new SATA SSDs.

> It is possible, of course. But if it was by UX and for everyone, I wouldn't be writing here.

Oh, well, it's unlikely to be a priority and so unlikely to be implemented ever, I think there were similar feature requests since multiple years ago. The only way of how this would appear is if someone interested (e.g. you) would implement this on their own apparently.

PS: give a reasonable name to the issue. Thanks.

**Comment by ilya-fedin** [comment]
Is have a setting is a must for you? If not, we can merge this with #107.

**Comment by guglovich** [comment]
> Well, there are new SATA SSDs.

Yes, of course. I just have to give some justification for my request.

> PS: give a reasonable name to the issue. Thanks.

Thank you, corrected.

> Is have a setting is a must for you? If not, we can merge this with #107.

Let it be separate. I need at least that.

**Comment by guglovich** [comment]
> Well, create a symlink for ~/.local/share/TelegramDesktop and move whenever you want?

An important clarification is that the path must be deep, otherwise you will have to make an authorisation every time.
/home/user/.local/share/TelegramDesktop/tdata/user_data/

And it's better to mount it, because /tmp is cleared, which is the main advantage.

**Comment by guglovich** [comment]
But in practice, both methods are still not working, only if you make the script run when Linux is switched on.

**Comment by mb6ockatf** [comment]
@guglovich I have HDD and the speed (at least. how telegram is running) is awesome.
Closed (crypted) config is something what provides encryption, i guess...
Why symlinks are not working? Make symlink for a folder and define any place for cache you want

**Comment by guglovich** [comment]
> 1. @guglovich I have HDD and the speed (at least. how telegram is running) is awesome.
> 
>     2. Closed (crypted) config is something what provides encryption, i guess...
> 
>     3. Why symlinks are not working? Make symlink for a folder and define any place for cache you want

I need the fastest drive, in this case RAM. And RAM is known to be cleared after boot, along with the symlinks in /tmp.
